### PR TITLE
Allow editing profile names

### LINKED ISSUE #1738: Allow editing username in profiles
**Problem**
What problem is this feature addressing?
Once we ship the Username feature there's not a way to edit the username

**Solution**
What's your proposed solution to the problem?
Build a way to edit the username from the profile

**Specification**
What precisely needs to get built? (include any requirements or design files)
Add an Edit user name button to the profile page

Part of #1395

### PR DESCRIPTION
**Pull request by sshmm**
## Motivation and Context
resolves #1738

## Description
1- changed editProfileForm to have user name input field and edited the schema for that
2 show error if the name is already in use

## Test and Deployment Plan
1- Go to profile page and click editprofile
2- update your name and click save

## Screenshots (if appropriate):

image

### CODE REVIEW

File: src/components/EditProfileModal.tsx [hunk 1]
```diff
@@ -141,11 +167,33 @@ export const EditProfileModal = ({
...
+              Profile Image
+            </Text>
+            <AvatarUpload original={myProfile.avatar} />
+          </Flex>
+          <Flex column css={{ alignItems: 'center' }}>
+            <Text p css={sectionHeader}>
+              Profile Name{' '}
+              <Tooltip
+                content={
+                  <div>You can&apos;t edit your name during active epochs</div>
```
Comment (CryptoGraffe) [inline src/components/EditProfileModal.tsx:182]: Is there consensus around this? What if you are ALWAYS in an active recurring epoch?

Comment (CryptoGraffe) [reply src/components/EditProfileModal.tsx:182]: Seems like this warning should only show during an active epoch

File: src/pages/ProfilePage/queries.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+              epochs: [
+                {
+                  where: {
+                    _and: [
+                      { ended: { _eq: false } },
+                      { start_date: { _lt: 'now' } },
+                      {
+                        circle: {
+                          organization: {
+                            name: { _neq: 'Sample DAO' },
```
Comment (CryptoGraffe) [inline src/pages/ProfilePage/queries.ts:25]: This should use the sample boolean instead of this string

File: src/pages/ProfilePage/queries.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                  },
+                },
+                { id: true },
+              ],
+            },
+          },
+        ],
+      },
+    ],
+  });
```
Comment (CryptoGraffe) [inline src/pages/ProfilePage/queries.ts:39]: would be nice to have an operationName in here

**Review by CryptoGraffe — COMMENTED** [review]
A couple minor comments here, with some more discussion in discord. Seems to work ok for me when testing.

File: src/components/EditProfileModal.tsx (hunk 1 above)
Comment (CryptoGraffe) [reply src/components/EditProfileModal.tsx:182]: Is this *also* enforced on the server side?

Comment (crabsinger) [reply src/components/EditProfileModal.tsx]: @sshmm did you see these?

Comment (crabsinger) [reply src/components/EditProfileModal.tsx]: 👍

**Review by crabsinger — APPROVED** [review]

**Pull request merged**